Lib/test/test_capi/test_watchers.py

```diff
@@ -351,12 +351,10 @@ def test_clear_unassigned_watcher_id(self):
             self.clear_watcher(1)
 
     def test_no_more_ids_available(self):
-        contexts = [self.watcher() for i in range(self.TYPE_MAX_WATCHERS)]
-        with ExitStack() as stack:
-            for ctx in contexts:
-                stack.enter_context(ctx)
-            with self.assertRaisesRegex(RuntimeError, r"no more type watcher IDs"):
-                self.add_watcher()
+        with self.assertRaisesRegex(RuntimeError, r"no more type watcher IDs"):
+            with ExitStack() as stack:
+                while True:
```

Comment:
Isn't it kinda awkward to while True just because we don't know how many watchers are registered?

Maybe limit this to "at most [max watchers]" instead of infinity?

Reply:
Sure, I can make that change.